### DISCUSSION TITLE
Eslint-TypeScript is really annoying, how to disable for non-TS files?

### DISCUSSION THREAD

**Discussion by DrakoPOD** [quasarframework/quasar#10880] [CLI - SSR mode]
I'm trying to test some libraries, so I have a test component, I don't added `lang="ts"` to the tag, but still I'm getting Eslint Error when I declare some variable.
`@typescript-eslint/no-unsafe-call: Unsafe call of an 'any' typed value.`

The component isn't even yet imported but it show the error on the console when I run in dev, the web page runs without any problem, because the component is not imported just created.

Is my first time using TS, but I need to use for this project. I already added `/* eslint-disable  */` but I still get the warning for this file.

**Comment by sschneider-ihre-pvs** [comment]
Do you have `allowJs: true` in the tsconfig?
you could add an override to eslint like:
```
overrides: [
    {
      files: ['*.js],
      rules: {
        '@typescript-eslint/no-unsafe-call: 'off'
      },
    },
]
```

**Reply by DrakoPOD** [reply]
AllowJs, only works for import `.js` files to `.ts` files. And my problem is with `.vue` files that don't have `lang="ts"` in the script tag.

**Comment by ontwikkelfabriek** [comment]
@DrakoPOD 

See this thread for solutions:
https://github.com/typescript-eslint/typescript-eslint/issues/1928

**Reply by DrakoPOD** [reply]
I read in other forum, this will disable for all vue files, even for those that have `ts` in their tag.

**Reply by ontwikkelfabriek** [reply]
@DrakoPOD 

> I read in other forum, this will disable for all vue files, even for those that have ts in their tag.

Ah I see. Wanting to do different things in eslint for the files with the same extension is a **very** uncommon thing... the only way I can think of is creating a some custom eslint rule...

**Reply by DrakoPOD** [reply]
I don't if is this a bug or something, if I try to put `ts` type to variables, if the vue file doesn't have `lang="ts"` if give me a warning `Only use typescript in ts files` but if I don't, give me `unsafe call of any`

**Comment by moHaHa** [comment]
is there any other updat ?
